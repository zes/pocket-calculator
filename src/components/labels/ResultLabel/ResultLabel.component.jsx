import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './ResultLabel.css';

export default class ResultLabel extends PureComponent {
  render() {
    const { result } = this.props;

    return <span className="result">{result}</span>;
  }
}

ResultLabel.propTypes = {
  result: PropTypes.string.isRequired,
};
