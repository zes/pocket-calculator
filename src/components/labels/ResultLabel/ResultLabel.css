.result {
  color: #ffffff;
  font-weight: bold;
}