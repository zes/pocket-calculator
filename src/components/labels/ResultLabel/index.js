export { default } from './ResultLabel.container';
