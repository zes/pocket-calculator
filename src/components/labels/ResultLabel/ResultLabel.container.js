import { connect } from 'react-redux';

import ResultLabel from './ResultLabel.component';

const mapStateToProps = state => ({
  result: state.result,
});

export default connect(mapStateToProps)(ResultLabel);
