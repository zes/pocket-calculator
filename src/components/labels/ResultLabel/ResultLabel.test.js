import React from 'react';
import renderer from 'react-test-renderer';
import ResultLabel from './ResultLabel.component';

describe('[Component] <ResultLabel />', () => {
  const props = {
    result: '84',
  };

  it('renders a snapshot', () => {
    const tree = renderer.create(<ResultLabel {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
