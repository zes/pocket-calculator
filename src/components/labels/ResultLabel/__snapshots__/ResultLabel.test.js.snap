// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] <ResultLabel /> renders a snapshot 1`] = `
<span
  className="result"
>
  84
</span>
`;
